crates/hdk3/src/entry.rs

```diff
@@ -1,4 +1,597 @@
-pub mod create_entry;
-pub mod delete_entry;
-pub mod hash_entry;
-pub mod update_entry;
+use crate::prelude::*;
+
+/// General function that can create any entry type.
+///
+/// This is used under the hood by [ `create_entry` ], [ `create_cap_grant` ] and [ `create_cap_claim` ].
+///
+/// The host builds a [ `Create` ] header for the passed entry value and commits a new element to the
+/// chain.
+///
+/// Usually you don't need to use this function directly; it is the most general way to create an
+/// entry and standardises the internals of higher level create functions.
+pub fn create(entry_def_with_id: EntryWithDefId) -> ExternResult<HeaderHash> {
+    host_call::<EntryWithDefId, HeaderHash>(__create, entry_def_with_id)
+}
+
+/// Update any entry type.
+///
+/// This is used under the hood by [ `update_entry` ], [ `update_cap_grant` ] and `update_cap_claim`.
+/// @todo implement update_cap_claim
+///
+/// The host builds an [ `Update` ] header for the passed entry value and commits a new update to the
+/// chain.
+///
+/// Usually you don't need to use this function directly; it is the most general way to update an
+/// entry and standardises the internals of higher level create functions.
+pub fn update(hash: HeaderHash, entry_with_def_id: EntryWithDefId) -> ExternResult<HeaderHash> {
+    host_call::<UpdateInput, HeaderHash>(__update, UpdateInput::new(hash, entry_with_def_id))
+}
+
+/// General that can delete any entry type.
+///
+/// This is used under the hood by [ `delete_entry` ], [ `delete_cap_grant` ] and `delete_cap_claim`.
+/// @todo implement delete_cap_claim
+///
+/// The host builds a [ `Delete` ] header for the passed entry and commits a new element to the chain.
+///
+/// Usually you don't need to use this function directly; it is the most general way to update an
+/// entry and standardises the internals of higher level create functions.
+pub fn delete(hash: HeaderHash) -> ExternResult<HeaderHash> {
+    host_call::<HeaderHash, HeaderHash>(__delete, hash)
+}
+
+/// Create an app entry.
+///
+/// An app entry is anything that the app can define a type for that matches the entry defs and
+/// that can be serialized.
```

Comment:
@maackle mmm i'm not sure, the intent of the sentence is to convey the broadest/loosest possible constraint on the data/abstraction that can be used for entries, because there is a common misconception that structs are 1:1 with entries